src/app/book/booklist/booklist.component.ts

```diff
@@ -0,0 +1,36 @@
+import { Component, OnInit } from '@angular/core';
+import { BookService } from '../book.service';
+import { Router } from '@angular/router';
+import { Book } from '../book';
+import { NgbRatingConfig } from '@ng-bootstrap/ng-bootstrap';
+
+@Component({
+  selector: 'app-booklist',
+  templateUrl: './booklist.component.html',
+})
+export class BooklistComponent implements OnInit {
+
+  constructor(private bookService: BookService,private config: NgbRatingConfig, private router : Router) { 
+    config.readonly = true;
+    config.max=5;
+  }
+
+  books : Book[] = [];
+
+  ngOnInit() {
+    let categoryid = 1;
```

Comment:
any reason for hard coded value ?

Reply:
passing category id as 1 to get books of all categories. will work on it and will try to do it in better way.

Reply:
Technically, you should pass 0 initially which means get books with all categories.

Reply:
done.